x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/action/apikey/GetApiKeyRequestTests.java

```diff
@@ -120,33 +123,59 @@ public void writeTo(StreamOutput out) throws IOException {
 
     public void testSerialization() throws IOException {
         final String apiKeyId = randomAlphaOfLength(5);
-        final boolean ownedByAuthenticatedUser = true;
-        GetApiKeyRequest getApiKeyRequest = GetApiKeyRequest.usingApiKeyId(apiKeyId, ownedByAuthenticatedUser);
         {
+            final GetApiKeyRequest getApiKeyRequest1 = GetApiKeyRequest.builder()
+                .ownedByAuthenticatedUser(true)
+                .apiKeyName(apiKeyId)
+                .build();
             ByteArrayOutputStream outBuffer = new ByteArrayOutputStream();
             OutputStreamStreamOutput out = new OutputStreamStreamOutput(outBuffer);
             out.setVersion(randomVersionBetween(random(), Version.V_7_0_0, Version.V_7_3_0));
-            getApiKeyRequest.writeTo(out);
+            getApiKeyRequest1.writeTo(out);
 
             InputStreamStreamInput inputStreamStreamInput = new InputStreamStreamInput(new ByteArrayInputStream(outBuffer.toByteArray()));
             inputStreamStreamInput.setVersion(randomVersionBetween(random(), Version.V_7_0_0, Version.V_7_3_0));
             GetApiKeyRequest requestFromInputStream = new GetApiKeyRequest(inputStreamStreamInput);
 
-            assertThat(requestFromInputStream.getApiKeyId(), equalTo(getApiKeyRequest.getApiKeyId()));
+            assertThat(requestFromInputStream.getApiKeyId(), equalTo(getApiKeyRequest1.getApiKeyId()));
             // old version so the default for `ownedByAuthenticatedUser` is false
             assertThat(requestFromInputStream.ownedByAuthenticatedUser(), is(false));
         }
         {
+            final GetApiKeyRequest getApiKeyRequest2 = GetApiKeyRequest.builder()
+                .apiKeyName(apiKeyId)
+                .ownedByAuthenticatedUser(true)
+                .withLimitedBy(true)
```

Comment:
I think it still makes sense to randomize here, i.e.:

```suggestion
                .withLimitedBy(randomBoolean())
```